### ISSUE TITLE
QGIS 3.4.10 crashes - Python 

### ISSUE THREAD

**Issue by Claudia500** [qgis/QGIS#31304]
Hi, could anybody help me?
I am working with QGIS 3.4.1 and when using addVectorLayer function at Python console, QGIS crashes with the following message

Crash ID: a0f76a7f494ae8e3c889f5e1cfe8faec9a73863f

Stack Trace
```
QgisApp::addVectorLayerPrivate :
QgisApp::addVectorLayer :
QgisAppInterface::addVectorLayer :
QgsPresetSchemeColorRamp::clone :
PyMethodDef_RawFastCallKeywords :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyEval_EvalCodeEx :
PyEval_EvalCode :
PyArena_Free :
PyRun_StringFlags :
PyImport_Import :
PyImport_Import :
PyMethodDef_RawFastCallKeywords :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyImport_Import :
PyImport_Import :
PyMethodDef_RawFastCallKeywords :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyFunction_FastCallDict :
PyMethodDef_RawFastCallDict :
PyObject_Call :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
QMetaObject::activate :
QAction::activate :
QAbstractButton::click :
QAbstractButton::mouseReleaseEvent :
QToolButton::mouseReleaseEvent :
QWidget::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
DispatchMessageW :
NotifyWinEvent :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main :
BaseThreadInitThunk :
RtlUserThreadStart :




QGIS Info
QGIS Version: 3.4.10-Madeira
QGIS code revision: 3d97aaae88
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 2.4.1
Running against GDAL: 2.4.1



System Info
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.3.9600
```

**Comment by m-kuhn** [comment]
What's the exact command you are executing?

**Comment by m-kuhn** [comment]
Please also try with the latest QGIS 3.4 release.

**Comment by Claudia500** [comment]
I'm working with command iface.addVectorLayer('layer path','layer name', 'OGR').
When I run this command, it crashes. Even working with QGIS 3.8.2 it crashes.

**Comment by nyalldawson** [comment]
We need more context. It's somewhere else in your script or process causing this.

**Comment by unron0830** [comment]
I met the same problem, too.

here is my code:
`fn = 'C:/temp/Taipei.shp'`
`vlayer = iface.addVectorLayer(fn,'C','org')`

when I execute the code and QGIS would crash.

But when I  execute  `iface.addRasterLayer`  it could display in the screen.

my QGIS ver :
The current available LTR version of QGIS is 3.4.11. 
The latest available version of QGIS is 3.8.2. 
We are working on 3.9-Master. It will go into feature freeze on 2019-08-30 12:00:00 UTC and be released as 3.10 on 2019-10-25 12:00:00 UTC.

**Comment by m-kuhn** [comment]
- Are you able to test this on the current LTR (3.10)?
- Are you able to share your file

The problem seems to be related to providing an invalid provider key, but it shouldn't crash (and does not crash here on current master)

**Comment by unron0830** [comment]
thanks, for reply 

here is my data
[https://drive.google.com/drive/folders/1kI5lvo-uwlXZQW75ZvL69jnkgeoYnsiM?usp=sharing](url)

I also tried on LTR(3.10) and the crash still happened.

**Comment by m-kuhn** [comment]
Does it work if you replace `'org'` with `'ogr'`?

**Comment by unron0830** [comment]
it work !! 
thanks a lot !!

**Comment by m-kuhn** [comment]
Is anyone able to reproduce a crash with an invalid provider key?

**Comment by gioman** [comment]
> Is anyone able to reproduce a crash with an invalid provider key?

@m-kuhn it crashes on 3.4, not on 3.10 (tested on Windows).

**Comment by gioman** [comment]
@m-kuhn sorry they both crashes.

**Comment by rouault** [comment]
Was fixed in master, but still current in stable release. Fixed per https://github.com/qgis/QGIS/pull/36751

**Comment by PeterPetrik** [comment]
does not crashes (fixed) on 3.10/3.16, closing
